src/main/java/org/HdrHistogram/AbstractHistogram.java

```diff
@@ -296,15 +296,15 @@ private void init(final long lowestDiscernibleValue,
          */
         final long largestValueWithSingleUnitResolution = 2 * (long) Math.pow(10, numberOfSignificantValueDigits);
 
-        unitMagnitude = (int) Math.floor(Math.log(lowestDiscernibleValue)/Math.log(2));
+        unitMagnitude = (int) (Math.log(lowestDiscernibleValue)/Math.log(2));
         unitMagnitudeMask = (1 << unitMagnitude) - 1;
 
         // We need to maintain power-of-two subBucketCount (for clean direct indexing) that is large enough to
         // provide unit resolution to at least largestValueWithSingleUnitResolution. So figure out
         // largestValueWithSingleUnitResolution's nearest power-of-two (rounded up), and use that:
         int subBucketCountMagnitude = (int) Math.ceil(Math.log(largestValueWithSingleUnitResolution)/Math.log(2));
-        subBucketHalfCountMagnitude = ((subBucketCountMagnitude > 1) ? subBucketCountMagnitude : 1) - 1;
-        subBucketCount = (int) Math.pow(2, (subBucketHalfCountMagnitude + 1));
+        subBucketHalfCountMagnitude = subBucketCountMagnitude - 1;
```

Comment:
`largestValueWithSingleUnitResolution` is `2` at the smallest, so `log_2` of that is at least `1`. This would make the `false` branch of the ternary only apply when `subBucketCountMagnitude` was already 1 (never 0).